tests/api/test_collection.py

```diff
@@ -0,0 +1,352 @@
+                }
+            }
+        }
+    """
+
+    mock_create_thumbnails = Mock(return_value=None)
+    monkeypatch.setattr(
+        ('saleor.dashboard.collection.forms.'
+         'create_collection_background_image_thumbnails.delay'),
+        mock_create_thumbnails)
+
+    name = 'test-name'
+    slug = 'test-slug'
+    variables = {
+        'name': name, 'slug': slug, 'isPublished': True}
+    response = staff_api_client.post_graphql(
+        query, variables, permissions=[permission_manage_products])
+    get_graphql_content(response)
+    assert mock_create_thumbnails.call_count == 0
+
+
+def test_update_collection(
+        monkeypatch, staff_api_client, collection, permission_manage_products):
+    query = """
+        mutation updateCollection(
+            $name: String!, $slug: String!, $id: ID!, $isPublished: Boolean!) {
+            collectionUpdate(
+                id: $id, input: {name: $name, slug: $slug, isPublished: $isPublished}) {
+                collection {
+                    name
+                    slug
+                }
+            }
+        }
+    """
+
+    mock_create_thumbnails = Mock(return_value=None)
+    monkeypatch.setattr(
+        ('saleor.dashboard.collection.forms.'
+         'create_collection_background_image_thumbnails.delay'),
+        mock_create_thumbnails)
+
+    collection_id = to_global_id('Collection', collection.id)
+    name = 'new-name'
+    slug = 'new-slug'
+    variables = {
+        'name': name, 'slug': slug, 'id': collection_id, 'isPublished': True}
+    response = staff_api_client.post_graphql(
+        query, variables, permissions=[permission_manage_products])
+    content = get_graphql_content(response)
+    data = content['data']['collectionUpdate']['collection']
+    assert data['name'] == name
+    assert data['slug'] == slug
+    assert mock_create_thumbnails.call_count == 0
+
+
+def test_update_collection_with_background_image(
+        monkeypatch, staff_api_client, collection, permission_manage_products):
+    query = """
+        mutation updateCollection(
+            $name: String!, $slug: String!, $id: ID!, $backgroundImage: Upload, $isPublished: Boolean!) {
+            collectionUpdate(
+                id: $id, input: {
+                    name: $name, 
+                    slug: $slug, 
+                    backgroundImage: $backgroundImage, 
+                    isPublished: $isPublished
+                }
+            ) {
+                collection {
+                    slug
+                }
+                errors {
+                    field
+                    message
+                }
+            }
+        }
+    """
+
+    mock_create_thumbnails = Mock(return_value=None)
+    monkeypatch.setattr(
+        ('saleor.dashboard.collection.forms.'
+         'create_collection_background_image_thumbnails.delay'),
+        mock_create_thumbnails)
+
+    collection_id = to_global_id('Collection', collection.id)
+    name = 'new-name'
```

Comment:
So I think those variables are not necessary as we are not doing anything with them apart from listing them inside of `variables`